tikv config: add in memory pessismitic lock size limit config (#18916)
pingcap/docs: dynamic-config.md

| `pessimistic-txn.wake-up-delay-duration` | The duration after which a pessimistic transaction is woken up |
| `pessimistic-txn.pipelined` | Determines whether to enable the pipelined pessimistic locking process |
| `pessimistic-txn.in-memory` | Determines whether to enable the in-memory pessimistic lock |

> | `pessimistic-txn.in-memory-peer-size-limit`               | Control the memory size limit of pessimistic locks for one region                                                                                                                                                                                  |
> | `pessimistic-txn.in-memory-instance-size-limit`           | Control the memory size limit of pessimistic locks for one TiKV instance                                                                                                                                                                           |

Reviewer: ```suggestion
| `pessimistic-txn.in-memory-peer-size-limit`               | Controls the memory usage limit for in-memory pessimistic locks in a Region                                                                                                                                                                                  |
| `pessimistic-txn.in-memory-instance-size-limit`           | Controls the memory usage limit for in-memory pessimistic locks in a TiKV instance                                                                                                                                                                           |
```
